### DISCUSSION TITLE
Add type arguments to generic types

### DISCUSSION THREAD

**Discussion by RobertCraigie** [encode/httpx#2181] [Potential Issue] [answered]
I use [pyright](https://github.com/microsoft/pyright) to type check my code. In strict mode it will report an error if a type is unknown or partially unknown. A type is considered unknown if one or some of it's parts or not specified, e.g.

```py
# a is unknown
def foo(a) -> None:
    ...

# b is partially unknown
def foo(b: dict) -> None:
    ...

# c is known
def foo(c: dict[str, str]) -> None:
    ...
```

In essence the reasoning behind these diagnostics is for documentation, performance and to ensure consistent behaviour between different tools. For more details see their documentation, [Typing Guidance for Python Libraries](https://github.com/microsoft/pyright/blob/main/docs/typed-libraries.md).

I encountered these diagnostics because of the missing type arguments for the `extensions` argument in `Client.get` although there appears to be other places that this is also relevant. You can find all the occurrences of unknown types by installing pyright and running `pyright --verifytypes httpx`.

**Comment by relsunkaev** [comment, marked as answer]
Got a [PR](https://github.com/encode/httpx/pull/2266) ready for this.
